instrumentation/cassandra/cassandra-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/cassandra/v4_0/CassandraNetworkAttributesGetter.java

```diff
@@ -8,12 +8,52 @@
 import com.datastax.oss.driver.api.core.cql.ExecutionInfo;
 import com.datastax.oss.driver.api.core.metadata.Node;
 import io.opentelemetry.instrumentation.api.semconv.network.NetworkAttributesGetter;
+import io.opentelemetry.instrumentation.api.semconv.network.ServerAttributesGetter;
 import java.net.InetSocketAddress;
 import java.net.SocketAddress;
 import javax.annotation.Nullable;
 
 final class CassandraNetworkAttributesGetter
-    implements NetworkAttributesGetter<CassandraRequest, ExecutionInfo> {
+    implements ServerAttributesGetter<CassandraRequest>,
+        NetworkAttributesGetter<CassandraRequest, ExecutionInfo> {
+
+  @Nullable
+  @Override
+  public String getServerAddress(CassandraRequest request) {
+    SocketAddress socketAddress =
+        request
+            .getSession()
+            .getMetadata()
+            .getNodes()
+            .values()
+            .iterator()
+            .next()
+            .getEndPoint()
+            .resolve();
```

Comment:
Here you are using the first node as the server address. How do you know that this was the node that was actually connected?

Reply:
nodes is a map.. it has 1 [key, value] pair. I don't know how to get it otherwise. do you have any suggestions? 
key is an UUID in this case. not sure how to check whether it was the right node.

Reply:
ServerAttributesGetter doesn't have access to response. 
I will try out the ExperimentalAttributesExtractor for Cassandra to access server.address and port through the response.

Reply:
@laurit  updated this PR with the new approach. please take a look and let me know your thoughts.